### DISCUSSION TITLE
Add support for sidecar containers

### DISCUSSION THREAD

**Discussion by Davidsod** [seaweedfs/seaweedfs#5122] [Ideas]
Hello,
We use Seaweedfs in k8s, and we deploy it with the repo helm chart.

In our use case, we'd like to have the ability to add a  sidecar container for logs & metrics shipping as an [agent-based](https://vector.dev/docs/setup/deployment/roles/#sidecar), as a [DaemonSet-based](https://vector.dev/docs/setup/deployment/roles/#daemon) shipper isn't doable in our use-case.

I was thinking of getting the same capabilities that already exist for [initcontainers](https://github.com/search?q=repo%3Aseaweedfs/seaweedfs%20initContainers&type=code) but for sidecars, similarly to how
 [bitnami/redis](https://github.com/bitnami/charts/tree/main/bitnami/redis) has for adding sidecar containers for instance.

Assuming this capability is indeed not supported in the helm or not achievable, we will happily make a PR to add them, but wanted to check if maybe I've missed it before starting, and also if you have a recommendation/suggestion on how to easily add these features.

Thanks!

**Comment by chrislusf** [comment]
Sounds nice!

**Comment by Davidsod** [comment]
Check this out: #5125
